src/components/NewClusterModal.js

```diff
@@ -332,11 +332,17 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
       onChange: onEnvChange,
       isSearchable: true,
       isClearable: false,
-      options: [{ label: 'JUPYTER ENVIRONMENTS', options: _.map(({ label, image }) => ({ label, value: image }), leoImages) },
+      options: [
         {
-          label: 'OTHER ENVIRONMENTS',
-          options: [{ label: 'Custom Environment', value: CUSTOM_MODE }, { label: 'Project-Specific Environment', value: PROJECT_SPECIFIC_MODE }]
-        }]
+          label: 'JUPYTER ENVIRONMENTS', options: _.map(({ label, image }) => ({ label, value: image }),
+            _.filter(({ isCommunity }) => !isCommunity, leoImages))
```

Comment:
Suggestion: I think you could use `_.groupBy` instead of two separate `_filter`s. That would make it more explicit that the two resulting sets are both disjoint and, together, still contain all elements.

Reply:
hmm... i like the idea, but in the .json file `isCommunity` is optional, and only specified when true, which I like since most images are not community maintained. Without that the groups are ['true', 'undefined'], and there doesn't seem to be a clean way to map the keys without obfuscating the intent further